code/modules/spells/spell_types/charge.dm

```diff
@@ -68,6 +69,9 @@
 					if(C.maxcharge <= 1) //Div by 0 protection
 						C.maxcharge = 1
 						burnt_out = 1
+				if(istype(C.loc,/obj/item/weapon/gun))
```

Comment:
The spell cycles through contents of items in hands, specifically looking for internal batteries.